### ISSUE TITLE
Questions about mask resizing

### ISSUE THREAD

**Issue by cats-food** [ali-vilab/VACE#24]
Thanks for the great work!

I have a little question about mask resizing operation.
https://github.com/ali-vilab/VACE/blob/9ea71b5bd29dec73faf1f66c67d57166dd0da265/vace/models/wan/wan_vace.py#L152

Specifically, the reshape operation of [1 F H W] -> [64  F/4  H/8  W/8], it seems that this will result in a 64-channel mask, why do we need to split the spatial dim to channel dim instead of directly resizing them spatially? 

```  python3

    def vace_encode_masks(self, masks, ref_images=None):
        if ref_images is None:
            ref_images = [None] * len(masks)
        else:
            assert len(masks) == len(ref_images)

        result_masks = []
        for mask, refs in zip(masks, ref_images):
            c, depth, height, width = mask.shape
            new_depth = int((depth + 3) // self.vae_stride[0])
            height = 2 * (int(height) // (self.vae_stride[1] * 2))
            width = 2 * (int(width) // (self.vae_stride[2] * 2))

            # reshape
            mask = mask[0, :, :, :]
            mask = mask.view(
                depth, height, self.vae_stride[1], width, self.vae_stride[1]
            )  # depth, height, 8, width, 8
            mask = mask.permute(2, 4, 0, 1, 3)  # 8, 8, depth, height, width
            mask = mask.reshape(
                self.vae_stride[1] * self.vae_stride[2], depth, height, width
            )  # 8*8, depth, height, width

            # interpolation
            mask = F.interpolate(mask.unsqueeze(0), size=(new_depth, height, width), mode='nearest-exact').squeeze(0)
```

**Comment by hanzhn** [comment]
It origins from [Flux Fill](https://huggingface.co/black-forest-labs/FLUX.1-Fill-dev). 

We believe this op maximizes the preservation of mask spatial details. However, no careful experiments have been done about that. It's just an intuitive guess.

**Comment by cats-food** [comment]
Thank you for the reply!
